pkg/api/persistentvolumeclaim/util_test.go

```diff
@@ -292,3 +293,144 @@ func TestAnyDataSourceFilter(t *testing.T) {
 		})
 	}
 }
+
+func TestDropAllocatedResources(t *testing.T) {
+	tests := []struct {
+		name     string
+		feature  bool
+		spec     *core.PersistentVolumeClaimSpec
+		oldSpec  *core.PersistentVolumeClaimSpec
+		expected *core.PersistentVolumeClaimSpec
+	}{
+		{
+			name:     "for:newPVC=hasfield,oldPVC=doesnot,featuregate=false; should drop field",
+			feature:  false,
+			spec:     withAllocatedResource("5G"),
+			oldSpec:  getPVCSpec(),
+			expected: getPVCSpec(),
+		},
+		{
+			name:     "for:newPVC=hasfield,oldPVC=doesnot,featuregate=true; should keep field",
+			feature:  true,
+			spec:     withAllocatedResource("5G"),
+			oldSpec:  getPVCSpec(),
+			expected: withAllocatedResource("5G"),
+		},
+		{
+			name:     "for:newPVC=hasfield,oldPVC=hasfield,featuregate=false; should keep field",
+			feature:  false,
+			spec:     withAllocatedResource("10G"),
+			oldSpec:  withAllocatedResource("5G"),
+			expected: withAllocatedResource("10G"),
+		},
+		{
+			name:     "for:newPVC=hasfield,oldPVC=nil,featuregate=false; should drop field",
+			feature:  false,
+			spec:     withAllocatedResource("5G"),
+			oldSpec:  nil,
+			expected: getPVCSpec(),
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.RecoverExpansionFailure, test.feature)()
+
+			DropDisabledFields(test.spec, test.oldSpec)
+
+			if !reflect.DeepEqual(*test.expected, *test.spec) {
+				t.Errorf("Unexpected change: %+v", diff.ObjectDiff(test.expected, test.spec))
+			}
+		})
+	}
+}
+
+func TestSetAllocatedResources(t *testing.T) {
+	tests := []struct {
+		name     string
+		feature  bool
+		spec     *core.PersistentVolumeClaimSpec
+		oldSpec  *core.PersistentVolumeClaimSpec
+		expected *core.PersistentVolumeClaimSpec
+	}{
+		{
+			name:     "feature:true; should default to requested size when crating new PVC",
+			feature:  true,
+			spec:     withResource(getPVCSpec(), "5G"),
+			oldSpec:  nil,
```

Comment:
I miss tests with `oldSpec` with capacity, but no AllocatedResources. One for capacity bigger than `spec` and one for smaller than `spec`.

Reply:
added tests